src/model/ExternalFile.cpp

```diff
@@ -136,6 +136,9 @@ namespace model {
     path ExternalFile_Impl::filePath() const {
       path result;
       path fname = toPath(fileName());
+      if (openstudio::filesystem::exists(fname) || openstudio::filesystem::is_regular_file(fname)) {
```

Comment:
Did you mean `&&` instead of `||`. The path must exist AND it must be a regular file no?

Reply:
I did mean `II`. I copied that line straight from below in the `absoluteFilePaths` loop.

Reply:
that doesn't make sense to me. try calling is_regular_file on an non existing path and it'll throw.
if it exists, then the is_regular_file is never called.

Reply:
bool is_regular_file(const path& p);

    Returns: is_regular_file(status(p)).

    Throws: filesystem_error if status(p) would throw filesystem_error.

https://www.boost.org/doc/libs/1_79_0/libs/filesystem/doc/reference.html#is_regular_file

Reply:
Is it wrong down below then, too?

Reply:
I guess it doesn't throw, but anyways...

https://www.boost.org/doc/libs/1_79_0/libs/filesystem/doc/reference.html#status

> Throws: filesystem_error. [Note: result values of file_status(file_not_found)and file_status(type_unknown) are not considered failures and do not cause an exception to be thrown. —end note]

Reply:
@joseph-robertson yes it's wrong below as well.